components/Main/Card.js

```diff
@@ -44,15 +47,61 @@ export default function MainPage() {
     }
   };
 
-  if (error) {
-    return (
-      <h1>
-        {error.message === "Recipe not found"
-          ? "Recipe not found"
-          : "failed to load "}
-      </h1>
-    );
-  }
+  const handleLike = async (recipe) => {
+    const updatedRecipe = {
+      ...recipe,
+      likes: recipe.likes + 1,
+    };
+
+    const response = await fetch(`/api/recipes/${recipe._id}`, {
+      method: "PUT",
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify(updatedRecipe),
+    });
+
+    if (response.ok) {
+      setFilteredRecipes((prevRecipes) =>
+        prevRecipes.map((prevRecipe) =>
+          prevRecipe._id === recipe._id
+            ? { ...prevRecipe, likes: recipe.likes + 1 }
```

Comment:
```suggestion
            ? { ...prevRecipe, likes: updatedRecipe.likes + 1 }
```
Shouldn't it be the updated recipe here?